pallets/runtime/common/src/runtime.rs

```diff
@@ -845,6 +845,19 @@ macro_rules! runtime_apis {
                 {
                     Asset::unsafe_can_transfer_granular(from_custodian, from_portfolio, to_custodian, to_portfolio, ticker, value)
                 }
+
+                #[inline]
+                fn balance_at(
+                    ticker: Ticker, checkpoint: CheckpointId, dids: Vec<IdentityId>
+                ) -> node_rpc_runtime_api::asset::BalanceAtResult
+                {
+                    let balances = dids
+                        .into_iter()
+                        .take(100) // max query size limit
+                        .map(|did| Asset::get_balance_at(ticker, did, checkpoint).into())
+                        .collect();
+                    Ok(balances)
```

Comment:
It  looks like an `infallible` method, Do we really need to return a `Result<_>`?

Reply:
`rpc_forward_call!` macro requires the return type to be a `Result`. It's either an `Ok` here or in `rpc/src/asset.rs`. So I just made a design decision and placed it here.